### DISCUSSION TITLE
Overwritten priority of tailwind classes

### DISCUSSION THREAD

**Discussion by eensander** [tailwindlabs/tailwindcss#11614] [Help]
I am using tailwindcss in a Vue 3 project. My `styles/main.scss` looks like this: 

```
@tailwind base;

@import 'minimal.scss';

@tailwind components;
@tailwind utilities;
```

`minimal.scss` contains general definitions for some elements, such as `a`, `input[type="text"]`, etc. The problem here is that when I am applying tailwind classes to an element in an component, they get overwritten by the definitions in minimal.scss. I would expect the priority of minimal.scss to be lower than utilities/components, since it is higher in the list. However, this seems to not be the case. 

Any suggestions or alternative solutions?

**Comment by wongjn** [comment]
You could consider reducing the specificity of your rules to `0 1 0` or lower since most Tailwind utilities are `0 1 0`.

For example, `input[type="text"]` has specificity `0 1 1` which would take precedence over any rule that only has a single class selector. Compare this to `:where(input[type="text"])` which has `0 0 0` and can be more easily overridden.

You could also consider organizing CSS into [native cascade `@layers`](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Cascade_layers).

**Reply by eensander** [reply]
Thanks for your reply. Interesting read, never realized how specificity was calculated. Though, I did disliked the `:where` solution. I was also unable to reliably get it to work using @layers, though this was my preferred solution. Eventually, I chose to invert the situation and added a specific class to al 'base' styles, which all elements with base styles should implement.
